apps/pano/app/routes/posts/__filters/my-posts.tsx

```diff
@@ -0,0 +1,25 @@
+import { useLoaderData } from "@remix-run/react";
+import type { LoaderFunction } from "react-router-dom";
+import { json } from "react-router-dom";
+import { authenticator } from "~/authenticator.server";
+import { PostList } from "~/features/post/PostList";
+import type { PostWithCommentCount } from "~/models/post.server";
+import { getMyPosts } from "~/models/post.server";
+import { User } from "~/models/user.server";
+
+type LoaderData = {
+  posts: PostWithCommentCount[];
+};
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const user = (await authenticator.isAuthenticated(request)) as User | null;
```

Comment:
we have an utility function named `requireUser` in `authenticator.server.tsx` which allows redirect to `/login` page and if that's not the case you'll be sure `user` will be there so you don't have to do a null check.

Reply:
That makes sense. I will make the necessary changes.